molecule/default/create.yml

```diff
@@ -15,94 +15,194 @@
+        rules:
+          - proto: tcp
+            ports: 22
+            cidr_ip: 0.0.0.0./0
+          - proto: tcp
+            ports: 80
+            cidr_ip: 0.0.0.0/0
+          - proto: tcp
+            ports: 443
+            cidr_ip: 0.0.0.0/0
+          - proto: tcp
+            ports: 5556
+            cidr_ip: 0.0.0.0/0
+            rule_desc: Allow traffic for using Dex as OIDC server + IDP
+        state: present
+      when: security_group.security_groups | length == 0
+      register: created_security_group
+
+    - name: Setting security group ID if it existed
+      ansible.builtin.set_fact:
+        sgID: "{{ security_group.security_groups[0]['group_id'] }}"
+      when: security_group.security_groups | length > 0
 
-    - name: Debug parsed testing-farm results
-      ansible.builtin.debug:
-        msg: "{{ tf_parsed_results }}"
-      when: debug_outputs
+    - name: Setting security group ID to new group
+      ansible.builtin.set_fact:
+        sgID: "{{ created_security_group.group_id }}"
+      when: security_group.security_groups | length == 0
 
-    # TODO: idempotency
-    - name: Populate instance config dict  # noqa jinja
+    - name: Generate instance ID
       ansible.builtin.set_fact:
-        instance_conf_dict: {
-          'instance': "{{ item.uuid }}",
-          'address': "{{ item.ip }}",
-          # TODO: testing-farm is giving us root by default, ideally we would get non-root to get conditions closer to prod environments
-          'user': "{{ item.username }}",
-          'port': "22", # TODO: get it dynamically
-          'identity_file': "{{ private_key_path  }}",
-        }
-      with_items: "{{ tf_parsed_results }}"
-      register: instance_config_dict
-
-    - name: Convert instance config dict to a list
+        instance_id: "{{ lookup('pipe', 'uuidgen') | regex_replace('[^a-zA-Z0-9]', '') | truncate(8) }}"
+
+    - name: Launch EC2 instance
+      amazon.aws.ec2_instance:
+        name: "{{ 'CI-' + (lookup('env', 'GITHUB_RUN_ID') if lookup('env', 'GITHUB_RUN_ID') | length > 0 else instance_id) }}"
+        key_name: "{{ aws_key_name }}"
+        instance_type: "{{ instance_type }}"
+        image_id: "{{ image_id }}" 
+        region: "{{ lookup('env', 'AWS_DEFAULT_REGION') }}"
+        vpc_subnet_id: "{{ subnet_id }}"
+        network:
+          assign_public_ip: true
+        security_group: "{{ sgID }}"
+        wait: yes
+        state: running
+      register: ec2_instance
+
+    - name: Set instance ID and IP
       ansible.builtin.set_fact:
-        instance_conf: "{{ instance_config_dict.results | map(attribute='ansible_facts.instance_conf_dict') | list }}"
-
-    - name: Debug instance_conf
-      ansible.builtin.debug:
-        msg: "{{ instance_conf }}"
-      when: debug_outputs
+        instanceID: "{{ ec2_instance.instances[0].instance_id}}"
+        instanceIP: "{{ ec2_instance.instances[0].network_interfaces[0].association.public_ip }}"
 
-    - name: Dump instance config
```

Comment:
I think it would be better to keep using the dumped instance config - that provides a way for molecule to understand whether the VM already exists or not. If we're ever going to use this locally for development, running `molecule create` twice wouldn't actually spawn 2 VMs, it would know that the VM is already spawned and inform the user of it.

You can also save the instance ID in the molecule instance config to access it in `destroy.yml`.